conans/test/unittests/tools/files_patch_test.py

```diff
@@ -184,11 +184,74 @@ def build(self):
         client.save({"conanfile.py": file_content})
         client.run("install .")
         client.run("build .", assert_error=True)
-        self.assertIn("patch: error: no patch data found!", client.out)
+        self.assertIn("patch_ng: error: no patch data found!", client.out)
         self.assertIn("ERROR: conanfile.py (test/1.9.10): "
                       "Error in build() method, line 12", client.out)
         self.assertIn("Failed to parse patch: string", client.out)
 
+    def test_add_new_file(self):
+        """ Validate issue #5320
+        """
+
+        conanfile = dedent("""
+            from conans import ConanFile, tools
+            import os
+            
+            class ConanFileToolsTest(ConanFile):
+                name = "foobar"
+                version = "0.1.0"
+                exports_sources = "*"
+            
+                def build(self):
+                    tools.patch(patch_file="add_files.patch")
+                    assert os.path.isfile("foo.txt")
+                    assert os.path.isfile("bar.txt")
```

Comment:
If you can assert the contents of one of them would be even nicer!

Reply:
done